line-bot-api-client/src/main/java/com/linecorp/bot/client/LineMessagingServiceImpl.java

```diff
@@ -0,0 +1,111 @@
+package com.linecorp.bot.client;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.linecorp.bot.model.PushMessage;
+import com.linecorp.bot.model.ReplyMessage;
+import com.linecorp.bot.model.profile.UserProfileResponse;
+import com.linecorp.bot.model.response.BotApiResponse;
+
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import okhttp3.ResponseBody;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+/**
+ * Proxy implementation of {@link LineMessagingService} to hind internal implementation (Retrofit as of 2016-10)
+ */
+@AllArgsConstructor(access = AccessLevel.PACKAGE)
+class LineMessagingServiceImpl implements LineMessagingService {
+    private static final String CONTENT_DISPOSITION_HEADER_NAME = "Content-Disposition";
+    private static final Pattern CONTENT_DISPOSITION_FNAME_PATTERN =
+            Pattern.compile("attachment; filename=\"(.*?)\"");
```

Comment:
This pattern matching is ugly. You should use a parser library implementation instead.